python_modules/dagster/dagster/_core/definitions/target.py

```diff
@@ -1,31 +1,25 @@
-from typing import TYPE_CHECKING, NamedTuple, Optional, Sequence, Union
+from typing import NamedTuple, Optional, Sequence, Union
 
 from typing_extensions import TypeAlias
 
 import dagster._check as check
 
+from .graph_definition import GraphDefinition
 from .job_definition import JobDefinition
 from .unresolved_asset_job_definition import UnresolvedAssetJobDefinition
 
-if TYPE_CHECKING:
-    from .graph_definition import GraphDefinition
-
 ExecutableDefinition: TypeAlias = Union[
     JobDefinition, "GraphDefinition", UnresolvedAssetJobDefinition
 ]
 
 
-class RepoRelativeTarget(NamedTuple):
-    """The thing to be executed by a schedule or sensor, selecting by name a job in the same repository."""
-
-    job_name: str
-    op_selection: Optional[Sequence[str]]
-
-
-class DirectTarget(
+class AutomationTarget(
     NamedTuple(
-        "_DirectTarget",
-        [("target", ExecutableDefinition)],
+        "_AutomationTarget",
+        [
+            ("job_resolvable", Union[ExecutableDefinition, str]),
```

Comment:
what does `job_resovable` mean?

Reply:
A piece of data that is resolvable to a `JobDefinition`. One of:

- `JobDefinition`
- `UnresolvedAssetJobDefinition`
- `GraphDefinition`
- str (a job name)

This property of being resolvable to a `JobDefinition` is what unites all of the entities that a `*Target` can wrap, both before and after this PR (I think this is the reason that `asset_selection` _does not_ get stored in a `*Target`, because it can't be resolved to as single job).

Reply:
Ok I think we should have a little helper class `ResolvableToJob` that contains this information as a doc block.